src/Illuminate/Notifications/Channels/MailChannel.php

```diff
@@ -100,8 +100,8 @@ protected function buildView($message)
         }
 
         return [
-            'html' => $this->markdown->render($message->markdown, $message->data()),
-            'text' => $this->markdown->renderText($message->markdown, $message->data()),
+            'html' => fn ($messageData = []) => $this->markdown->render($message->markdown, $messageData),
```

Comment:
It looks like theme is respected if I write `return (new MailMessage)->markdown('markdown')->theme('abba-zabba');` within the toMail() method of the notification.

That said, the instance level $markdown should be cloned, since we are changing the theme. Though Markdown is not bound as a singleton in the container, subsequent calls would use the same theme, even tho the user would anticipate it to be the result.

Reply:
But can you adjust the code, so the "set" of theme happens exactly before the rendering? Just like the code I've done for the other PR?

Reply:
@nunomaduro take a look at how I have implemented it with the latest 2 commits. I can move it into the closure if that's better, but I don't see any way that the message's theme property would be overridden. (Also, I just think it would look uglier).

Could we remove the Markdown instance variable and just create it within the closures (or `buildView()` method)? It would mean a signature change to the constructor, but that instance level is more of a hinderance than a help, IMO.